routers/repo/branch.go

```diff
@@ -1,32 +1,144 @@
 // Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2017 The Gitea Authors. All rights reserved.
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
 package repo
 
 import (
-	"code.gitea.io/gitea/modules/base"
+	"time"
+
+	"code.gitea.io/git"
 	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/log"
 )
 
 const (
-	tplBranch base.TplName = "repo/branch"
+	tplBranchOverview = "repo/branches/overview"
+	tplBranchAll      = "repo/branches/all"
 )
 
-// Branches render repository branch page
-func Branches(ctx *context.Context) {
-	ctx.Data["Title"] = "Branches"
-	ctx.Data["IsRepoToolbarBranches"] = true
+// Branch branch information on UI
+type Branch struct {
+	Name        string
+	Commit      *git.Commit
+	IsProtected bool
+}
 
-	brs, err := ctx.Repo.GitRepo.GetBranches()
+func loadBranches(c *context.Context) []*Branch {
```

Comment:
Could we name it `ctx` for consistency with other handlers? Likewise in other places